### PR TITLE
local-gadget: Add trace sni gadget

### PR DESCRIPTION
**Pull request by mqasimsarfraz**
#893

### CODE REVIEW

File: cmd/local-gadget/trace/sni.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	commonutils "github.com/inspektor-gadget/inspektor-gadget/cmd/common/utils"
+	"github.com/inspektor-gadget/inspektor-gadget/cmd/local-gadget/utils"
+	containercollection "github.com/inspektor-gadget/inspektor-gadget/pkg/container-collection"
+	containerutils "github.com/inspektor-gadget/inspektor-gadget/pkg/container-utils"
+	sniTracer "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/sni/tracer"
+	sniTypes "github.com/inspektor-gadget/inspektor-gadget/pkg/gadgets/trace/sni/types"
+	localgadgetmanager "github.com/inspektor-gadget/inspektor-gadget/pkg/local-gadget-manager"
+)
+
+const logalGadgetSubKey = "local-gadget-key"
```
Comment (mauriciovasquezbernal) [inline cmd/local-gadget/trace/sni.go:36]: logalGadgetSubKey -> localGadgetSubKey :smile:

**Review by mauriciovasquezbernal — COMMENTED** [review]
I'd like to first discuss on https://github.com/inspektor-gadget/inspektor-gadget/pull/1103 the approach for enriching the data and then we can decide how to move forward.

**Comment by mauriciovasquezbernal** [comment]
https://github.com/inspektor-gadget/inspektor-gadget/pull/1103 was merged. @mqasimsarfraz would you mind updating this one please?

File: cmd/local-gadget/trace/sni.go [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+				event.Namespace = container.Namespace
+				event.Pod = container.Podname
+				event.Container = container.Name
+			}
+
+			switch commonFlags.OutputMode {
+			case commonutils.OutputModeJSON:
+				b, err := json.Marshal(event)
+				if err != nil {
+					fmt.Fprint(os.Stderr, fmt.Sprint(commonutils.WrapInErrMarshalOutput(err)))
```
Comment (mauriciovasquezbernal) [inline cmd/local-gadget/trace/sni.go:80]: nit: we're prepending "Error: " when printing those error messages -> https://github.com/inspektor-gadget/inspektor-gadget/pull/1103#discussion_r1026474208 (probably there was another discussion about it but I can't find it)

File: cmd/local-gadget/trace/sni.go [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
+					return
+				}
+
+				fmt.Println(string(b))
+			case commonutils.OutputModeColumns:
+				fallthrough
+			case commonutils.OutputModeCustomColumns:
+				fmt.Println(parser.TransformIntoColumns(&event))
+			default:
+				fmt.Fprint(os.Stderr, commonutils.WrapInErrOutputModeNotSupported(commonFlags.OutputMode))
```
Comment (mauriciovasquezbernal) [inline cmd/local-gadget/trace/sni.go:90]: We don't need to check this case anymore -> https://github.com/inspektor-gadget/inspektor-gadget/pull/1135

**Review by mauriciovasquezbernal — APPROVED** [review]
Two small comments but LGTM otherwise.

**Comment by mqasimsarfraz** [comment]
@mauriciovasquezbernal thanks for the review. Do you have want to have final look? I updated the PR to address the nits. I will merge this once CI is green.

**Comment by mauriciovasquezbernal** [comment]
It's good. It can be merged once CI is green.

**Pull request merged**